src/rpn/rpn.ts

```diff
@@ -1,31 +1,57 @@
 export function rpn(inputString: string): any {
+    console.log(inputString)
     if (inputString.length === 420) throw new Error("Blaze it");
+    if ( /[a-zA-Z]/i.test(inputString)) throw new Error("Invalid Expression");
 
-    const operandsAndOperators: Array<number | string> = inputString.split(" ").map((token) => {
-        var parsedToken = isNaN(Number(token))
+    const operandsAndOperators: Array<number | string> = inputString.split(" ").map((token: string) => {
+        return isNaN(Number(token))
             ? token
             : Number(token);
-        console.log(parsedToken)
-        return parsedToken;
     });
-    console.log(operandsAndOperators)
 
-    const stack: number[] = [];
+    if (operandsAndOperators
+        .filter(x => typeof x === "number")
+        .length === operandsAndOperators
+        .filter(x => typeof x === "string")
+        .length + 1) {
 
-    operandsAndOperators.forEach((operandOrOperator) => {
-        let result;
-
-        if (typeof operandOrOperator === "string") {
-            // @ts-ignore
-            result = ((a: number, b: number) => a + b)(...stack.splice(-2));
-        } else result = operandOrOperator;
-        stack.push(result);
-    });
+        const stack: number[] = [];
 
+        operandsAndOperators.forEach((operandOrOperator: number | string) => {
```

Comment:
I approached it from a different angle. To simplify the logic, I brought my code closer to the algorithm sketch I saw at the bottom.
1. I check if the token is a number.
- I put it on the stack (or not)
2. If I hit the operator,
- I check what he is like
- I do calculations
- I put the result on the stack
etc.

'm using the pop method to make it clearer.
The exclamation point next to the variable tells the compiler that the variable is correct. I have this certainty because I validate at the very beginning.
In my opinion, this notation is simpler, the result variable can be eliminated, and the whole thing describes the algorithm better.